main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -45,30 +45,42 @@ object Simplifier {
         val fs = spec.fparams.map(visitFormalParam)
         val exp = visitExp(impl.exp)
         val funType = impl.inferredScheme.base
-        val retType = funType.arrowResultType
-        val eff = funType.arrowEffectType
+        val retType = visitType(funType.arrowResultType)
+        val eff = visitType(funType.arrowEffectType)
         SimplifiedAst.Def(spec.ann, spec.mod, sym, fs, exp, retType, eff, sym.loc)
     }
 
     /**
       * Translates the given expression `exp0` to the SimplifiedAst.
       */
     def visitExp(exp0: LoweredAst.Expr): SimplifiedAst.Expr = exp0 match {
-      case LoweredAst.Expr.Var(sym, tpe, loc) => SimplifiedAst.Expr.Var(sym, tpe, loc)
+      case LoweredAst.Expr.Var(sym, tpe, loc) =>
+        val t = visitType(tpe)
+        SimplifiedAst.Expr.Var(sym, t, loc)
 
-      case LoweredAst.Expr.Def(sym, tpe, loc) => SimplifiedAst.Expr.Def(sym, tpe, loc)
+      case LoweredAst.Expr.Def(sym, tpe, loc) =>
+        val t = visitType(tpe)
+        SimplifiedAst.Expr.Def(sym, t, loc)
 
-      case LoweredAst.Expr.Cst(cst, tpe, loc) => SimplifiedAst.Expr.Cst(cst, tpe, loc)
+      case LoweredAst.Expr.Cst(cst, tpe, loc) =>
+        val t = visitType(tpe)
+        SimplifiedAst.Expr.Cst(cst, t, loc)
 
       case LoweredAst.Expr.Lambda(fparam, exp, tpe, loc) =>
         val p = visitFormalParam(fparam)
         val e = visitExp(exp)
-        SimplifiedAst.Expr.Lambda(List(p), e, tpe, loc)
+        visitType(tpe) match {
```

Comment:
I understand the desire to do this, but we simply don't.

If we were to do it, we would do it consistently (e.g. everywhere for all expressions), but that turns out to be too much work.

Instead we simply use Type everywhere (MonoType) and that has worked out fine.

Reply:
So I request to drop these special cases for Lambda and just use MonoType everywhere in SimplifiedAst.

Reply:
Then we just need to do it in a later phase

Reply:
We do? Where do we require Type.Lambda? I think it should just be `Type`?

Reply:
In `LambdaLift.scala` we need to access `tpe.result`. Would it make more sense to handle it there?

Reply:
Keep it local